app/paid/validator.go

```diff
@@ -0,0 +1,83 @@
+package paid
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/dgrijalva/jwt-go"
+)
+
+type pubKeyManager struct {
+	url string
+	key *rsa.PublicKey
+}
+
+var pubKM *pubKeyManager
```

Comment:
something about these singleton instances feels wrong. all its really doing is storing an rsa key (the `url` is only used in one place to download the key). 

this whole file could be replaced by two functions: `Load([]byte) *rsa.PublicKey` to load the key from bytes, and `Validate(key *rsa.PublicKey, token string) (*StreamToken, error)` to validate the token using the key. then you load the key elsewhere (maybe in InstallRoutes or some other setup function) and put it on the requests that need it

Reply:
But the logic of downloading the key from a certain url still has to reside somewhere. Doesn't feel like routes file or any of the initialization files are great for this purpose

Reply:
I agree that they don't feel quite right though.

Reply:
I think I'm actually at peace with those. They are singletons containing some persistent state that cannot be stored in any other way and some logic about that state that only gets called by two publicly exposed functions. I think this is the easiest way to go about it in a way that would make for easy re-use of this in lbrytv-player code 

:change_my_mind.jpg:

Reply:
is it weird that the pubkey manager is downloading a key from the internet? feels like it shouldn't be responsible for that. instead of passing a URL to Init(), i'd pass is the data (or an io.Reader).

Reply:
I see your point. Agreed